xgi/utils/utilities.py

```diff
@@ -79,3 +79,35 @@ def powerset(
 
     s = list(iterable)
     return chain.from_iterable(combinations(s, r) for r in range(start, len(s) + end))
+
+
+def update_uid_counter(H):
+    """
+    Helper function to make sure the uid counter is set correctly after
+    adding edges with different methods.
+
+    If we don't set the start of self._edge_uid correctly, it will start at 0,
+    which will overwrite any existing edges when calling add_edge().  First, we
+    use the somewhat convoluted float(e).is_integer() instead of using
+    isinstance(e, int) because there exist integer-like numeric types (such as
+    np.int32) which fail the isinstance() check.
+
+    Parameters
+    ----------
+    H : xgi.Hypergraph
+        Hypergraph of which to update the uid counter
+
+    """
+
+    edges_with_int_id = [
```

Comment:
The only concern that I have with this is that it would be amazing if we could somehow leverage the `self._edge_uid` to make this an O(1) update instead of an O(|E|) update. As in, once we validate the edges that have been added to update `self._edge_uid`, we shouldn't need to iterate over these edges again when we update again.

Reply:
I fully agree. I couldn't think of way now though. Should we merge and raise an issue to improve this?

Reply:
If you looked into it an it turned out to not be straightforward then yes let's just open a new issue.

Reply:
I have an idea now but might not have time to implement it in the next days. I'll raise the issue.